File: GraphInput.scala
object GraphInput{
    def graphInput(fname: String): DirectedGraph = {
        import scala.io.Source

        val file = Source.fromFile(fname).getLines
        val n = file.length

        var m = new Array[Array[Int]](n)
        var j = 0

        for (line <- Source.fromFile(fname).getLines) {
            var row = line.split(", ")
            var rowMat = new Array[Int](n)
            var i = 0
            for(elem <- row){
                if(elem == "MAX"){
                    rowMat(i) = Int.MaxValue
                } else{
                    rowMat(i) = elem.toInt
                }
                i += 1
            }
            m(j) = rowMat
            j += 1
        }

        return DirectedGraph(m)
    }
}

File: Graph Drawer/GraphInput.py
def graphInput(fname):
    f = open(fname)
    data = [ x.rstrip("\n") for x in f.readlines() ]

    mat = []

    for row in data:
        rowMat = []
        rowSplit = row.split(" ")
        for elem in rowSplit:
            if elem == 'MAX':
                rowMat.append("MAX")
            else:
                rowMat.append(int(elem))
        mat.append(rowMat)

    return mat